File: load_products.py
import firebase_admin
from firebase_admin import credentials, firestore
import random

# Inicializar Firebase
cred = credentials.Certificate("/home/winchester/Vídeos/Ptoyecto App/price-compare-app/serviceAccountKey.json")
firebase_admin.initialize_app(cred)
db = firestore.client()

# Limpiar la colección 'products' antes de cargar nuevos productos
docs = db.collection('products').stream()
for doc in docs:
    doc.reference.delete()
print("Productos existentes eliminados.")

# Lista de categorías y tiendas para generar datos realistas
categories = ["Bebidas", "Cocina", "Limpieza", "Cuidado Personal", "Snacks", "Lácteos", "Panadería", "Congelados"]
stores = ["Supermercado A", "Supermercado B", "Supermercado C", "Tienda D", "Hipermercado E"]

# Lista base de productos para generar variaciones
base_products = [
    ("Leche Entera 1L", 900, "Lácteos"),
    ("Yogur Natural 500g", 600, "Lácteos"),
    ("Pan Lactal 500g", 450, "Panadería"),
    ("Arroz Integral 1kg", 800, "Cocina"),
    ("Aceite de Oliva 500ml", 1200, "Cocina"),
    ("Detergente Líquido 1L", 700, "Limpieza"),
    ("Shampoo 400ml", 850, "Cuidado Personal"),
    ("Galletitas Dulces 200g", 350, "Snacks"),
    ("Jugo de Naranja 1L", 500, "Bebidas"),
    ("Helado de Vainilla 1L", 1100, "Congelados"),
    ("Pasta Fideos 500g", 400, "Cocina"),
    ("Salsa de Tomate 500g", 300, "Cocina"),
    ("Jabón en Polvo 800g", 650, "Limpieza"),
    ("Acondicionador 400ml", 900, "Cuidado Personal"),
    ("Papas Fritas 150g", 400, "Snacks"),
    ("Agua Mineral 2L", 250, "Bebidas"),
    ("Pizza Congelada 400g", 950, "Congelados"),
    ("Manteca 200g", 500, "Lácteos"),
    ("Harina 1kg", 350, "Cocina"),
    ("Esponja de Limpieza", 150, "Limpieza"),
]

# Generar 200 productos variando precios y tiendas
products = []
for i in range(200):
    base_product = base_products[i % len(base_products)]
    name = base_product[0]
    base_price = base_product[1]
    category = base_product[2]
    # Variar el precio entre -20% y +20% y redondear a 2 decimales
    price_variation = random.uniform(0.8, 1.2)
    price = round(base_price * price_variation, 2)
    store = random.choice(stores)
    products.append({
        "name": name,
        "price": price,
        "store": store,
        "category": category
    })

# Cargar los productos a Firestore
for product in products:
    db.collection('products').add(product)
    print(f"Producto agregado: {product['name']} - ARS {product['price']} ({product['store']})")

print("Carga de productos completada!")

File: loadProducts.mjs
import { initializeApp } from 'firebase/app';
import { getFirestore, collection, addDoc } from 'firebase/firestore';
import { db } from './firebase.js';

// Nombres reales de productos (30 nombres, se repetirán para llegar a 150)
const productNames = [
  'Leche Entera La Serenísima 1L',
  'Yogur Natural Danone 500g',
  'Pan Lactal Fargo 500g',
  'Queso Cremoso La Paulina 400g',
  'Manteca Sancor 200g',
  'Arroz Integral Lucchetti 1kg',
  'Fideos Spaghetti Matarazzo 500g',
  'Aceite de Girasol Cocinero 900ml',
  'Azúcar Ledesma 1kg',
  'Harina 000 Morixe 1kg',
  'Café Molido La Virginia 250g',
  'Té en Saquitos Taragüi 50u',
  'Galletitas Dulces Criollitas 300g',
  'Mermelada de Frutilla Arcor 454g',
  'Atún en Lomitos La Campagnola 170g',
  'Salsa de Tomate Cica 520g',
  'Mayonesa Natura 500g',
  'Ketchup Hellmann’s 250g',
  'Jabón Líquido Ala 900ml',
  'Detergente Magistral 750ml',
  'Esponja de Cocina Virulana 2u',
  'Papel Higiénico Elite 4 rollos',
  'Shampoo Sedal 400ml',
  'Acondicionador Dove 200ml',
  'Jabón en Barra Lux 90g',
  'Pasta Dental Colgate 90g',
  'Cerveza Brahma 1L',
  'Gaseosa Coca-Cola 1.5L',
  'Agua Mineral Villavicencio 2L',
  'Jugo de Naranja Cepita 1L',
];

// Categorías y tiendas
const categories = ['Cocina', 'Limpieza', 'Bebidas'];
const stores = ['Supermercado A', 'Supermercado B', 'Tienda C'];

async function loadProducts() {
  // Generar 150 productos (repetimos nombres para simular productos iguales en diferentes tiendas)
  for (let i = 0; i < 150; i++) {
    const name = productNames[i % productNames.length]; // Cicla entre los nombres
    const product = {
      name: name,
      price: Math.floor(Math.random() * 1000) + 100, // Precio entre 100 y 1100
      store: stores[Math.floor(Math.random() * stores.length)],
      category: categories[Math.floor(Math.random() * categories.length)],
      name_lower: name.toLowerCase(), // Para búsqueda
    };
    try {
      await addDoc(collection(db, 'products'), product);
      console.log(`Producto "${name}" cargado`);
    } catch (error) {
      console.error(`Error cargando producto "${name}":`, error);
    }
  }
  console.log('Carga de 150 productos completada');
}

loadProducts();